pages/signUp.tsx

```diff
@@ -1,37 +1,91 @@
-import React from 'react'
+import React, { ChangeEvent, useState } from 'react'
 import NavBar from '@/components/NavBar'
+import { useRouter } from 'next/router'
 
+function signUp(props) {
+
+  const [email, setEmail] = useState()
+  const [confirmEmail, setConfirmEmail] = useState()
+  const [password, setPassword] = useState()
+  const [confirmPassword, setConfirmPassword] = useState()
+  const [type, setType] = useState('')
+  const [toolTip, setToolTip] = useState(false)
+  const [alert, setAlert] =useState('')
+  const router = useRouter()
+  
 
-const callSignUpApi = async () => {
+  const callSignUpApi = async () => {
     const params = {
-      test: 'test'
+      email: email,
+      password: password
     }
     const options = {
       method: 'POST',
       body: JSON.stringify(params),
     }
     const data = await fetch('/api/signup', options)
-    
-    console.log(await data.json())
+    const dataText = await data.text()
+    if(dataText === 'User made successfully!'){
+      router.push('/')
+    }
+    else if(dataText === "User already exists!"){
+      setAlert("User already exists!")
+    }
+    else{
+      setAlert('Unkown error occured, please try again later')
+    }
+    console.log(dataText)
+    console.log(typeof data.body)
   }
 
+  function handleEmailChange(event) {
+    setEmail(event.target.value)
+  }
+  function handleConfirmEmailChange(event) {
+    setConfirmEmail(event.target.value)
+  }
+  function handlePasswordChange(event) {
+    setPassword(event.target.value)
+  }
+  function handleConfirmPasswordChange(event) {
+    setConfirmPassword(event.target.value)
```

Comment:
See above what I said about the repeated functionality - we need a more reusable function